### PR TITLE
Added Localization Support + German (de)

### PR DESCRIPTION
**Pull request by jonnius**

### CODE REVIEW

**Comment by HarlemSquirrel** [comment]
Thanks for your contribution! Sorry but I did not get a notification about this PR and just saw it now. Have you tried this out to make sure it is working?

**Comment by jonnius** [comment]
Yes I did. But one should rebase it to make sure it is still working. I may only find time for it in a month.

File: locale/emoji-buckets.pot [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2017-04-13 11:59+0200\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
```
Comment (HarlemSquirrel) [inline locale/emoji-buckets.pot]: These should probably be filled in.

**Review by HarlemSquirrel — CHANGES_REQUESTED** [review]
Looks pretty good here just one thing below. Let me know when you've had a chance to rebase and test again. Thanks!

File: locale/emoji-buckets.pot (hunk 1 above)
Comment (jonnius) [reply locale/emoji-buckets.pot]: This is the auto generated template file. No need to fill in these fields there.

**Comment by jonnius** [comment]
I have rebased the branch on your master branch. It is working fine.

File: convenience.js [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
```
Comment (HarlemSquirrel) [inline convenience.js:27]: Where does this copyright come from?

File: locale/emoji-buckets.pot [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+msgid "Preferences..."
+msgstr ""
+
+#: prefs.js:26
+msgid "Font size"
+msgstr ""
+
+#: prefs.js:26
+msgid "requires restart"
+msgstr ""
```
Comment (HarlemSquirrel) [inline locale/emoji-buckets.pot:62]: I'm not sure why we need this file along with `emoji-buckets.po`

**Review by HarlemSquirrel — CHANGES_REQUESTED** [review]

File: locale/emoji-buckets.pot (hunk 3 above)
Comment (jonnius) [reply locale/emoji-buckets.pot:62]: It is the starting point for new translations. Run ```xgettext -k_ -kN_ -o locale/emoji-buckets.pot *.js --from-code=UTF-8``` to update it automatically from the code. There are developers separating po files (sources) from mo files (binaries) and provide a makefile or a script to build the translation (create the mo files) automatically. See [Dash to Panel](https://github.com/home-sweet-gnome/dash-to-panel) e.g. But even they check in the pot file for translators convenience.

File: convenience.js (hunk 2 above)
Comment (jonnius) [reply convenience.js:27]: I have copied it from some other extension. There are a lot of extensions using this file. You find the authors name and mail address at the top of the file.

**Review by HarlemSquirrel — APPROVED** [review]

**Comment by HarlemSquirrel** [comment]
Alright, let's merge this guy.

**Pull request merged**